campaign-launcher/interfaceV2/src/providers/Web3AuthProvider.tsx

```diff
@@ -0,0 +1,130 @@
+import {
+  createContext,
+  FC,
+  PropsWithChildren,
+  useContext,
+  useState,
+  useEffect,
+} from 'react';
+
+import { useAccount, useSignMessage } from 'wagmi';
+
+import { recordingApi } from '../api';
+import { REFRESH_FAILURE_EVENT } from '../api/recordingApiClient';
+import { tokenManager } from '../utils/TokenManager';
+
+type Web3AuthContextType = {
+  isAuthenticated: boolean;
+  isLoading: boolean;
+  signIn: () => Promise<void>;
+  logout: () => Promise<void>;
+};
+
+const Web3AuthContext = createContext<Web3AuthContextType>(
+  {} as Web3AuthContextType
+);
+
+export const Web3AuthProvider: FC<PropsWithChildren> = ({ children }) => {
+  const [isAuthenticated, setIsAuthenticated] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const { signMessageAsync } = useSignMessage();
+  const { isConnected, address } = useAccount();
+
+  const signIn = async () => {
+    setIsLoading(true);
+    try {
+      const nonce = await recordingApi.getNonce(address);
+      const signature = await signMessageAsync({
+        message: JSON.stringify(nonce),
+      });
+      const authResponse = await recordingApi.auth(address, signature);
+
+      tokenManager.setTokens({
+        access_token: authResponse.access_token,
+        refresh_token: authResponse.refresh_token,
+      });
+      setIsAuthenticated(true);
+    } catch(e) {
+      setIsAuthenticated(false);
+      console.error('Failed to sign in');
```

Comment:
Since it's not an AxiosError anymore
```suggestion
      console.error('Failed to sign in', e);
```